bosch_locator_bridge/src/rosmsgs_datagram_converter.cpp

```diff
@@ -593,3 +593,72 @@ std::vector<uint64_t> RosMsgsDatagramConverter::readSensorOffsets(Poco::BinaryRe
 
   return sensor_offsets;
 }
+
+size_t RosMsgsDatagramConverter::convertClientExpandMapVisualizationDatagram2Message(
+      const std::vector<char>& datagram,
+      bosch_locator_bridge::ClientExpandMapVisualization& client_expandmap_visualization)
+{
+    Poco::MemoryInputStream inStream(&datagram[0], datagram.size());
+    auto binary_reader = Poco::BinaryReader(inStream, Poco::BinaryReader::LITTLE_ENDIAN_BYTE_ORDER);
+    binary_reader.setExceptions(std::ifstream::failbit | std::ifstream::badbit | std::ifstream::eofbit);
+
+    double stamp;
+    binary_reader >> stamp;
+    client_expandmap_visualization.timestamp = ros::Time(stamp);
+    binary_reader >> client_expandmap_visualization.visualization_id;
+
+    // Get zones
+    uint32_t zones_length;
+    binary_reader >> zones_length;
+    for (unsigned int i = 0; i < zones_length; i++)
+    {
+      bosch_locator_bridge::ClientExpandMapOverwriteZoneInformation zone;
+      uint32_t polygon_length;
+      binary_reader >> polygon_length;
+      for (unsigned int i = 0; i < polygon_length; i++)
+      {
+        float pose_x, pose_y;      
+        binary_reader >> pose_x >> pose_y;
+        geometry_msgs::Point point;
+        point.x = static_cast<double>(pose_x);
+        point.y = static_cast<double>(pose_y);
+        zone.polygon.push_back(point);
+      }
+
+      binary_reader >> zone.id >> zone.type;
+
+      uint32_t name_length;
+      binary_reader >> name_length;
+      std::vector<char> name(name_length);
```

Comment:
Can this not be created as a string directly e.g.:
```suggestion
      std::string name(name_length);
```

Reply:
I want to initialize the size of the string with not yet known content.